### DISCUSSION TITLE
Observing nested composited value

### DISCUSSION THREAD

**Discussion by AvivonCtrl** [mobxjs/mobx#3468] [Q&A] [answered]
Given the following snippet inside a `superObject`:
```typescript
class SuperObject {
   this.myFields = []
   nestedObject = new NestedObject({myNestedFields: this.myFields})
}
```

Alongside an observer component that is rendering `superObject.nestedObject.myNestedFields`

I update the super model (`superObject`, the model that composites `nestedObject` in itself) like so ->
```typescript
this.myFields = [{"name":"test"}]
```

Which allegedly also changes the inside model's `myNestedFields` -- I would expect the component to re-render and to be aware of the top-level changes that also propagates to the composited `nestedObject`, however - it is not the case.

Is this the expected behavior or am I missing something fundamental? Feels to me like an OOTB, regular observation use-case.

Should be noted I solved it using the `reaction` approach which I'm less fond of and am looking for a better way:

```typescript
this.fieldsDisposer = reaction(
      () => this.myFields,
      (myFields) => {
        this.nestedObject.setFields(...myFields);
      },
    );|
```

**Comment by mweststrate** [comment, marked as answer]
Did you check https://mobx.js.org/understanding-reactivity.html and the difference between `autorun` and `reaction`? 

The object model you are describing is a bit hard to follow, I recommend to create a code sandbox instead to clarify your problem.

**Reply by AvivonCtrl** [reply]
Yep, I'm claiming the `reaction` approach works for me, but I'd rather opt for a solution without it.
I'll try making it as-pseudo-as-possible cause it can be exemplified in 3 rows:
```typescript
class SuperObject {
   this.myFields = []
   nestedObject = new NestedObject({myNestedFields: this.myFields})
   this.myFields = [1, 2, 3]
}
```
In this case, observing `myNestedFields` does not cause a re-render.

**Reply by urugator** [reply]
`this.myFields = [1, 2, 3]` does not modify `myNestedFields` (you can check this out by simply printing the content of `myNestedFields` after the assigment), so why it should re-render?
You can modify the array itself, eg `this.myField.replace([1,2,3])`

**Reply by AvivonCtrl** [reply]
Assuming `this.myFields` is a getter that manipulates `this.foo`, running `this.foo.replace([1,2,3])` still doesn't propagate.

**Reply by AvivonCtrl** [reply]
Ended up solving it by passing `this` to the nested object rather than the array :)

**Comment by mweststrate** [comment]
There are no observables. Or observers. We can't advice on what you might
be doing wrong without some code showing what you actually do.
